src/plus/integrations/authentication/jira.ts

```diff
@@ -1,111 +1,70 @@
-import type { CancellationToken, Disposable } from 'vscode';
+import type { CancellationToken, Disposable, Uri } from 'vscode';
 import { CancellationTokenSource, window } from 'vscode';
-import { Logger } from '../../../system/logger';
-import { cancellable } from '../../../system/promise';
+import type { Container } from '../../../container';
+import type { DeferredEventExecutor } from '../../../system/event';
+import { promisifyDeferred } from '../../../system/event';
 import { openUrl } from '../../../system/utils';
-import type { ServerConnection } from '../../gk/serverConnection';
+import { IssueIntegrationId } from '../providers/models';
+import type { CloudIntegrationsApi } from './cloudIntegrationsApi';
 import type {
 	IntegrationAuthenticationProvider,
 	IntegrationAuthenticationSessionDescriptor,
 } from './integrationAuthentication';
 import type { ProviderAuthenticationSession } from './models';
 
-type ConnectionType = 'oauth' | 'personal_access_token';
-
-type ProviderTokenData = {
-	accessToken: string;
-	type: ConnectionType;
-	domain: string;
-	expiresIn: number;
-	scopes: string;
-};
-
-type ProviderAuthorizationData = {
-	url: string;
-};
-
-/* type ProviderRsp<T> = {
-	data: T | null;
-	error: string | null;
-}; */
-
-type providerInput = 'jira' | 'trello' | 'gitlab' | 'github' | 'bitbucket' | 'azure';
-
 export class JiraAuthenticationProvider implements IntegrationAuthenticationProvider {
-	constructor(private readonly connection: ServerConnection) {}
+	constructor(
+		private readonly container: Container,
+		private readonly getCloudIntegrationsApi: () => Promise<CloudIntegrationsApi>,
```

Comment:
If we move the access to the `Container` we can avoid having to bind & pass this here